src/orion/core/worker/trial.py

```diff
@@ -527,7 +528,7 @@ def compute_trial_hash(
             ignore_experiment = True
 
         experiment_repr = ""
-        if not ignore_experiment:
+        if not ignore_experiment and trial.experiment is not None:
```

Comment:
Assuming that `trial.experiment : Optional[str]`:

```suggestion
        if not ignore_experiment and trial.experiment:
```

Reply:
Nop, it can be `str` or `int` depending on the DB backend. Thus `trial.experiment == 0` is possible.